packages/experimental/Badge/src/CounterBadge/CounterBadge.tsx

```diff
@@ -28,12 +28,13 @@ export const CounterBadge = compose<CounterBadgeType>({
       const { count, icon, iconPosition = 'before', overflowCount, dot, ...mergedProps } = mergeProps(badge.props, final);
       const { showBadge } = badge.state;
       const displayCount = count && count > overflowCount ? `${overflowCount}+` : `${count}`;
+
       return showBadge ? (
         <Slots.root {...mergedProps}>
           {!dot && (
             <React.Fragment>
               {icon && iconPosition === 'before' && <Slots.icon {...iconProps} />}
-              <Slots.text>{displayCount}</Slots.text>
+              {!Children.toArray(children)[0] && <Slots.text>{displayCount}</Slots.text>}
```

Comment:
Can we assign this check to a variable as it may not be apparent what you're trying to check for?